src/main/java/hudson/plugins/ec2/EC2AbstractSlave.java

```diff
@@ -386,7 +387,7 @@ public boolean getStopOnTerminate() {
      * Called when the slave is connected to Jenkins
      */
     public void onConnected() {
-        // Do nothing by default.
+        isConnected = true;
```

Comment:
Looks like it is used here https://github.com/jenkinsci/ec2-plugin/pull/186/files#diff-b1df2806507eaf67657940f500845f0bR476